src/Microsoft.Azure.WebJobs.Logging.ApplicationInsights/ApplicationInsightsLogger.cs

```diff
@@ -512,13 +529,28 @@ private void StartTelemetryIfFunctionInvocation(IDictionary<string, object> stat
             }
         }
 
-        private void PopulateLinks(Activity[] activities, RequestTelemetry request)
+        private IOperationHolder<RequestTelemetry> CreateRequestFromLinks(Activity[] activities, string functionName)
         {
-            if (activities == null)
+            var operation = _telemetryClient.StartOperation<RequestTelemetry>(functionName);
+            var request = operation.Telemetry;
+            // if there is more than one link (batch dispatch in EventHub trigger)
+            // we'll populate link information on the request telemetry, but don't touch parent for this request
+            PopulateLinks(activities, request);
+
+            // If any of the links is sampled in (on upstream) we also preliminary sample in 
+            // request telemetry and everything that happens in this request scope. 
+            // There will be additional level of sampling applied to limit rate to one 
+            // configured on this Function/WebJob
+            if (request.ProactiveSamplingDecision == SamplingDecision.SampledIn)
             {
-                return;
+                Activity.Current.ActivityTraceFlags |= ActivityTraceFlags.Recorded;
```

Comment:
What does this do? I feel like i asked this before, sorry :-)

Reply:
this solves problem of sampling for batched execution. Normally, when no batching is involved, we do best effort to make consistent sampling decisions based on the operation_Id, but when batching is involved, we generate a new one.

E.g. if there are 10 messages received and context in one of messages indicates that it was sampled in on the upstream service, we want to increase chances of  the batched request to be sampled in. 

So this flag tells AI SDK to try its best to sample in everything under this Activity, at the same time SDK does some post-processing that will keep telemetry rate at the configured threshold.